modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -754,6 +800,29 @@ private void shutdownScheduler() {
     }
   }
 
+  private int statusToNotificationType(PollContext.PollItemStatus status, boolean currentPollItemLimitApplied) {
+    if (currentPollItemLimitApplied) {
+      return ITEM_REJECTED_LIMIT;
+    }
+    switch (status) {
+      case ACCEPTED:
+        return ITEM_DISPATCHED;
+      case FILTERED_BY_WATERMARK:
+        return ITEM_REJECTED_WATERMARK;
+      case ALREADY_IN_PROCESS:
+        return ITEM_REJECTED_IDEMPOTENCY;
+      case SOURCE_STOPPING:
+        return ITEM_REJECTED_SOURCE_STOPPING;
+    }
+    return NO_ACTION_ID;
+  }
+
+  private void dispatchPollingSourceNotification(Notification notification) {
```

Comment:
remove this method. It's only purpose is to perform the if, but you already instantiated the notification, loosing cycles and GC pressure on collecting instances that will be most of the time discarded.